### DISCUSSION TITLE
Run a lightweight self-hosted instance

### DISCUSSION THREAD

**Discussion by hellodword** [anyproto/contributors#209] [Self-hosting]
See https://github.com/hellodword/anytype-all#self-hosted

---

[The official dockercompose](https://github.com/anyproto/any-sync-dockercompose) is good, but a bit too heavy for me, and the scripts are complicated:

```
$ docker compose stats --no-stream --format "table {{.Name}}\t{{.MemUsage}}"
NAME                                                 MEM USAGE
any-sync-dockercompose-netcheck-1                    19.05MiB
any-sync-dockercompose-any-sync-node-1-1             31.32MiB
any-sync-dockercompose-any-sync-filenode-1           43.5MiB
any-sync-dockercompose-any-sync-consensusnode-1      33.27MiB
any-sync-dockercompose-any-sync-coordinator-1        35.17MiB

any-sync-dockercompose-generateconfig-processing-1   20.54MiB
any-sync-dockercompose-generateconfig-anyconf-1      25.71MiB

any-sync-dockercompose-any-sync-node-3-1             31.77MiB
any-sync-dockercompose-any-sync-node-2-1             32MiB
any-sync-dockercompose-minio-1                       183.1MiB
any-sync-dockercompose-mongo-1-1                     308.5MiB
any-sync-dockercompose-redis-1                       26.3MiB
```
By applying some modifications and workarounds, I can run a self-hosted instance without Minio and MongoDB, using only one sync node:

```
$ docker compose stats --no-stream --format 'table {{.Name}}\t{{.CPUPerc}}\t{{.MemUsage}}'
NAME                                   CPU %     MEM USAGE / LIMIT
anytype-all-any-sync-node-1-1          0.00%     13.29MiB / 128MiB
anytype-all-any-sync-filenode-1        0.00%     12.88MiB / 128MiB
anytype-all-any-sync-consensusnode-1   0.00%     8.09MiB / 128MiB
anytype-all-redis-1                    0.16%     3.934MiB / 128MiB
anytype-all-any-sync-coordinator-1     0.00%     9.477MiB / 128MiB
```

This is the directory size within a new space:

```
$ du --max-depth=1 -h ./storage
20K	./storage/any-sync-coordinator
1.1M	./storage/redis
12K	./storage/any-sync-consensusnode
2.0M	./storage/any-sync-filenode
716K	./storage/any-sync-node-1
3.7M	./storage
```

It's highly personalized, **use at your own risk**.

---

Please let me know if you have questions or suggestions.

**Comment by jestarray** [comment]
Yeah, my vps has 1GB ram and ~600MB are in use for all my other stuff... Looks like mongo and minio  setup adds 581MB so it would go overboard unless I upgrade the vps :/ . Great job on making it lighter, I wish anytype used sqlite or something instead

**Reply by grishy** [reply]
If you want a simpler setup, I created a self-hosted Anytype bundle based on original codebase - https://github.com/grishy/any-sync-bundle

I only removed MinIO, the most consuming component and merge other to reuse components to make it lighter

**Reply by hellodword** [reply]
> If you want a simpler setup, I created a self-hosted Anytype bundle based on original codebase - https://github.com/grishy/any-sync-bundle
> 
> I only removed MinIO, the most consuming component and merge other to reuse components to make it lighter

I’ve known about this project for a while, and I’m glad to see it’s still active. I really like the approach, but I’ve never quite had the courage to deep dive into the source code of all the any* components. Great work!

**Reply by grishy** [reply]
I have a really light version - one binary with almost no RAM load.
But I decided not to merge it because it would be quite a lot of work to support in the future to be aligned with upstream.

https://github.com/grishy/any-sync-bundle/pull/19 - here I almost recreated, with tests and checks, all the storage and communication parts...
One restriction was only to be fully compatible with Proto API

**Comment by hellodword** [comment]
> I wish anytype used sqlite or something instead

I think it's possible with this FR https://github.com/anyproto/any-sync-coordinator/issues/80
